core/main/src/broker/websocket_broker.rs

```diff
@@ -18,79 +18,200 @@
             }
+
             loop {
-                tokio::select! {
+                tokio::select!(
                     Some(value) = &mut read => {
                         match value {
                             Ok(v) => {
                                 if let tokio_tungstenite::tungstenite::Message::Text(t) = v {
                                     // send the incoming text without context back to the sender
-                                    Self::handle_response(t.as_bytes(),callback.clone())
+                                    if let Err(e) = BrokerOutputForwarder::handle_non_jsonrpc_response(
+                                        t.as_bytes(),
+                                        callback_c.clone(),
+                                        request_c.clone(),
+                                    ) {
+                                        error!("error forwarding {}", e);
+                                    }
                                 }
                             },
                             Err(e) => {
                                 error!("Broker Websocket error on read {:?}", e);
-                                break false
+                                break;
                             }
                         }
 
                     },
                     Some(request) = tr.recv() => {
-                        debug!("Got request from receiver for broker {:?}", request);
-                        if let Ok(updated_request) = Self::update_request(&request) {
-                            debug!("Sending request to broker {}", updated_request);
-                             let _feed = ws_tx.feed(tokio_tungstenite::tungstenite::Message::Text(updated_request)).await;
+                        debug!("Recieved cleaner request for {}", request);
+                        if request.eq(&app_id) {
+                            let _feed = ws_tx.feed(tokio_tungstenite::tungstenite::Message::Close(None)).await;
                             let _flush = ws_tx.flush().await;
+                            break;
                         }
-
                     }
-                }
+                )
             }
         });
-        Self { sender: broker }
+        tx
     }
 }
 
+async fn connect(endpoint: &str, alias: Option<String>) -> (url::Url, TcpStream) {
+    info!("Broker Endpoint url {}", endpoint);
+    let url_path = if let Some(a) = alias {
+        format!("{}{}", endpoint, a)
+    } else {
+        endpoint.to_owned()
+    };
+    let url = url::Url::parse(&url_path).unwrap();
+    let port = extract_tcp_port(endpoint);
+    info!("Url host str {}", url.host_str().unwrap());
+    //let tcp_url = url.host_str()
+    let tcp = loop {
+        if let Ok(v) = TcpStream::connect(&port).await {
+            break v;
+        } else {
+            error!("Broker Wait for a sec and retry {}", port);
+            tokio::time::sleep(Duration::from_secs(1)).await;
```

Comment:
What if the url is not valid or the connection request is refused by server ?

Reply:
Websocket Brokers are intended for critical internal websockets so Connect is expected to succeed we keep retrying till it does based on the current expectation. This is the current strategy for the first phase of this implementation we can make this more sophisticated only if necessary. URL is configured on the rules file like any bad configuration will have impacts. In this case that particular API call will indefinitely wait.